zipline/pipeline/loaders/earnings_estimates.py

```diff
@@ -236,13 +253,185 @@ def get_requested_quarter_data(self,
         # dates so that we have a value for each calendar date.
         return requested_qtr_data.unstack(SID_FIELD_NAME).reindex(dates)
 
+    def get_split_adjusted_asof_idx(self, dates):
+        """
+        Compute the index in `dates` where the split-adjusted-asof-date
+        falls. This is the date up to which, and including which, we will
+        need to unapply all adjustments for and then re-apply them as they
+        come in. After this date, adjustments are applied as normal.
+
+        Parameters
+        ----------
+        dates : pd.DatetimeIndex
+            The calendar dates over which the Pipeline is being computed.
+
+        Returns
+        -------
+        split_adjusted_asof_idx : int
+            The index in `dates` at which the data should be split.
+        """
+        split_adjusted_asof_idx = dates.searchsorted(
+            self._split_adjusted_asof
+        )
+        # The split-asof date is after the date index.
+        if split_adjusted_asof_idx == len(dates):
+            split_adjusted_asof_idx = len(dates) - 1
+        elif self._split_adjusted_asof < dates[0].tz_localize(None):
+            split_adjusted_asof_idx = -1
+        return split_adjusted_asof_idx
+
+    def collect_overwrites_for_sid(self,
+                                   group,
+                                   dates,
+                                   requested_qtr_data,
+                                   last_per_qtr,
+                                   sid_to_idx,
+                                   columns,
+                                   all_adjustments_for_sid,
+                                   sid):
+        """
+        Given a sid, collect all overwrites that should be applied for this
+        sid at each quarter boundary.
+
+        Parameters
+        ----------
+        group : pd.DataFrame
+            The data for `sid`.
+        dates : pd.DatetimeIndex
+            The calendar dates for which estimates data is requested.
+        requested_qtr_data : pd.DataFrame
+            The DataFrame with the latest values for the requested quarter
+            for all columns.
+        last_per_qtr : pd.DataFrame
+            A DataFrame with a column MultiIndex of [self.estimates.columns,
+            normalized_quarters, sid] that allows easily getting the timeline
+            of estimates for a particular sid for a particular quarter.
+        sid_to_idx : dict[int -> int]
```

Comment:
Not a major issue (and wouldn't block on it) but at this level, I think passing the `idx` directly makes sense. Leaving the lookup to scope where this method is called.